robotoff/insights/annotate.py

```diff
@@ -875,6 +875,116 @@ def select_nutrition_image(
+        is_vote: bool = False,
+    ) -> AnnotationResult:
+        """Process image orientation annotation and rotate the selected image
+        if necessary.
+
+        :param insight: the image orientation insight
+        :param data: additional data sent by the client, defaults to None
+        :param auth: authentication data
+        :param is_vote: True if the annotation was triggered by an anonymous
+            vote, defaults to False
+        :return: annotation result
+        """
+        product_id = insight.get_product_id()
+        product = get_product(product_id, ["images"])
+
+        if product is None:
+            return MISSING_PRODUCT_RESULT
+
+        if insight.source_image is None:
+            logger.warning(
+                "Missing source_image in image orientation insight: %s", insight.id
+            )
+            return INVALID_DATA
+
+        image_id = get_image_id(insight.source_image)
+
+        if image_id is None:
+            logger.warning(
+                "Invalid image ID for image orientation insight: %s, %s",
+                image_id,
+                insight.id,
+            )
+            return INVALID_DATA
+
+        rotation = insight.data.get("rotation")
+
+        if rotation is None or rotation == 0:
+            logger.warning(
+                "Missing or zero rotation angle in image orientation insight: %s",
+                insight.id,
+            )
+            return INVALID_DATA
+
+        # We'll check if the image is a selected image by iterating through
+        # the product images looking for keys that indicate selected images
+        is_selected = False
+        selected_key = None
+        for key, image_data in product["images"].items():
+            if key.startswith(("front", "ingredients", "nutrition", "packaging")):
+                if image_data.get("imgid") == image_id:
+                    is_selected = True
+                    selected_key = key
+                    break
+
+        if not is_selected or selected_key is None:
+            logger.info(
+                "Image %s is not a selected image, not applying rotation", image_id
+            )
+            return ALREADY_ANNOTATED_RESULT
+
+        image_data = product["images"][selected_key]
+
+        # Extract current crop information if it exists
+        crop_bounding_box = None
+        if all(k in image_data for k in ("x1", "y1", "x2", "y2")):
+            # Get the full image size
+            size = image_data.get("sizes", {}).get("full", {})
+            height = size.get("h")
+            width = size.get("w")
+
+            if height and width:
+                # Get current coordinates
+                x1, y1 = float(image_data["x1"]), float(image_data["y1"])
+                x2, y2 = float(image_data["x2"]), float(image_data["y2"])
+
+                # Convert to relative coordinates if they're not already (0-1 range)
+                if x1 > 1 or y1 > 1 or x2 > 1 or y2 > 1:
```

Comment:
Coordinates in Product Opener are always in absolute coordinates. Also, `select_rotate_image` expects coordinates in the same format as Product Opener (which means absolute coordinates), so you shouldn't perform normalization here.